File: backend/src/main/java/com/phoneecommerce/backend/entity/CartItem.java
package com.phoneecommerce.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Data
@Table(name= "CartItem")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "Product_id")
    private Product item_id;

    private String name;

    private String brand;

    private double price;

    private String storage;

    private String ram;

    private Integer quantity;
}

File: frontend/src/types/CartItem.ts
/**
 * CartItem represents a product in the shopping cart
 * Extends product data with quantity information
 */
export interface CartItem {
  productId: number;
  name: string;
  brand?: string;
  price: number;
  imageUrl?: string;
  quantity: number;
  storage?: string;
  ram?: string;
  stock: number;
}

/**
 * Cart state structure
 */
export interface Cart {
  items: CartItem[];
  totalPrice: number;
  totalItems: number;
}
